third-party/llvm/README

```diff
@@ -1,14 +1,72 @@
-==========================
-LLVM README for Chapel
-==========================
+=======================
+LLVM for Chapel release
+=======================
+
+This copy of LLVM 7.0.1 is being released with Chapel for
+convenience and was obtained from
+
+https://www.llvm.org/
+
+Three Chapel-related patches were added to this copy
+of the code, and the LLVM test subdirectories were deleted.
+
+Any Chapel issues that seem to be related to LLVM should be directed
+to the Chapel team at https://chapel-lang.org/bugs.html.
+
+Using LLVM with Chapel
+======================
 
 Chapel can be built (by setting CHPL_LLVM=llvm) to include LLVM
-(http://llvm.org) in order to enable extern block support and LLVM code
-generation.
+in order to enable extern block support and LLVM code generation.
 
 For more information on the current support for LLVM within Chapel,
 please refer to $CHPL_HOME/doc/rst/technotes/llvm.rst.  For more
 information about LLVM itself, please refer to the website above or to
-the README in the llvm/ subdirectory of this directory.
+the README in the llvm-src/ subdirectory of this directory.
+
+Chapel modifications to LLVM
+============================
+
+The modifications that we have made to the official LLVM release are
+as follows:
+
+* BasicAliasAnalysis.cpp
+   - the change we apply is https://reviews.llvm.org/D38499
+   - that is subsumed by https://reviews.llvm.org/D38662
+     which is merged in r350220
+
+* ValueTracking.cpp
+   - this is https://reviews.llvm.org/D38501 which is merged in r350395
+
+* Linux.cpp
+   - this is https://reviews.llvm.org/D53770 which is merged in r346802
+
+* The llvm-src/test and llvm-src/tools/clang/test directories were
+  deleted because we do not use them.  Their deletion saves
+  significant space and checkout time.
+
+Upgrading LLVM versions
+=======================
+
+The directory $CHPL_HOME/third-party/llvm/llvm-src/ contains the
+un-tarballed LLVM package contents.  Version updates should be done as
+follows, assuming CWD is $CHPL_HOME/third-party/llvm/:
 
-The version of LLVM used here is 6.0 with two patches.
+1.  un-tarball the new LLVM version into the directory it specifies,
+    for example llvm-7.0.1.src
+2.  un-tarball the new Clang version into the directory it specifies,
+    for example cfe-7.0.1.src
+3.  mv cfe-7.0.1.src llvm-7.0.1.src/tools/clang
+4.  git rm -r llvm-src
+5.  mv llvm-7.0.1.src llvm-src
+6.  rm -r llvm-src/test llvm-src/tools/clang/test
+7.  Remove the line that says "tools/clang" from llvm-src/.gitignore
+8.  git add --force llvm-src
+    ('--force' is needed so that the 'git add' ignores our .gitignore)
```

Comment:
I'm confused about this comment - wouldn't we want llvm-src not to be .gitignore'd ?

Reply:
Right, it would probably be better for me just to say "to ensure git adds all files in the subdirectory".  I'll do that.